pkg/storage/ingest/fetcher.go

```diff
@@ -500,7 +501,9 @@ func (r *concurrentFetchers) run(ctx context.Context, wants chan fetchWant, logg
 			if f.Err != nil {
 				w = handleKafkaFetchErr(f.Err, w, errBackoff, r.startOffsets, r.client, attemptSpan)
 			}
-
+			if hwm := f.HighWatermark; hwm >= 0 {
+				casHWM(highWatermark, hwm)
+			}
```

Comment:
Wouldn't be safer to do it only if `f.Err == nil`? However, `casHWM()` ensures that we always cas to the highest value, and in case there was an error and `f.HighWatermark` is the zero value, then it's probably not a big deal.

Reply:
i wanted to always use the value from the response if it's there even if there's an error. It's not set if it's -1 (the library ensures this IIRC). And as you said a 0 isn't a big deal (we do fetches a couple of times a second, so an additional atomic load doesn't sound like a lot of overhead